style: apply /style-guide pass to models/sweeps (#2722)
wandb/docs: models/sweeps/signal-handling-sweep-runs.mdx

This applies whether the process exits from a signal handler, from an exception, or from an explicit `sys.exit()` call. Understanding and relying on this contract matters in preemptible or cluster environments.

When the process exits due to a [catchable signal](#catchable-signals-and-preemption), your handler can run, call [`wandb.run.mark_preempting()`](/models/ref/python/experiments/run#mark_preempting) if you want the run requeued, perform cleanup (for example, save a checkpoint), then exit with a non-zero code. A common convention is `sys.exit(128 + signum)` for termination by signal. W&B records that exit code and the same [requeue rules](/models/runs/resuming#resume-preemptible-sweeps-runs) apply. When the operating system kernel kills the process with [`SIGKILL`](#sigkill-uncatchable), the process can't run exit hooks, so W&B doesn't write a final summary and the run might appear as crashed or killed. The agent still starts the next run.

## Stale runs and server-side timeouts

> Exit codes aren't the only way W&B determines run state. The W&B server also infers run state from activity. If a run neither finishes nor posts new metrics for about 5 minutes, the W&B server marks the run as crashed. That often happens when the training process becomes unresponsive, stops logging, or terminates without a clean exit (for example, after `SIGKILL`). Logging metrics on a steady cadence or exiting with a defined code helps keep run state aligned with what happened.

Reviewer: ```suggestion
W&B determines run state both from exit codes and from the run's activity. If a run neither finishes nor posts new metrics for about 5 minutes, W&B marks the run as crashed. This can occur when the training process becomes unresponsive, stops logging, or terminates without a clean exit (such as by sending `SIGKILL`). Log metrics on a steady cadence or exit with a defined code to help keep run state aligned with what happened.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Catchable signals and preemption

> Most signals you'll encounter in preemptible environments are catchable, meaning your training script can intercept them and shut down cleanly. You can register custom signal handlers in your training script. When the system delivers a catchable signal, your handler runs. W&B preserves metrics that were already sent, and the agent detects the process exit and starts the next run.

Reviewer: ```suggestion
Most signals in preemptible environments are catchable, meaning your training script can intercept them and shut down cleanly. You can register custom signal handlers in your training script. When the system delivers a catchable signal, your handler runs. W&B preserves metrics it already received. The agent detects the process exit and starts the next run.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## `SIGKILL` (uncatchable)

> `SIGKILL` can't be caught or ignored. The process terminates immediately with no chance to run handlers or atexit callbacks. W&B can't write a final summary for the run. The agent still recovers and continues the sweep, but run data for that run is incomplete. Use `SIGKILL` only as a last resort. Prefer `SIGTERM` or `SIGINT` when you need graceful shutdown.

Reviewer: ```suggestion
`SIGKILL` comes from the operating system kernel  and can't be caught or ignored. The process terminates immediately with no chance to run handlers or `atexit` callbacks. W&B can't write a final summary for the run. The agent still recovers and continues the sweep, but run data for that run is incomplete. Use `SIGKILL` only as a last resort. Prefer `SIGTERM` or `SIGINT` when you need graceful shutdown.
```
Not sure what `atexit` is, some kind of a callout?

Author: Applied your suggestion (fixed the double space). `atexit` is Python's standard-library module for registering exit callbacks; it's now formatted as code.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
